packages/database/realtime/database.service.ts

```diff
@@ -0,0 +1,280 @@
+import {Injectable} from "@nestjs/common";
+import {ChangeStream, DatabaseService, Document, FilterQuery} from "@spica-server/database";
+import {Observable, Subject, Subscription} from "rxjs";
+import {bufferTime, filter, share} from "rxjs/operators";
+import {levenshtein} from "./levenshtein";
+import {late} from "./operators";
+import {ChunkKind, StreamChunk} from "./stream";
+
+@Injectable()
+export class RealtimeDatabaseService {
+  _streamCount = 0;
+
+  constructor(private database: DatabaseService) {}
+
+  find<T extends Document = any>(
+    name: string,
+    options: FindOptions<T> = {}
+  ): Observable<StreamChunk<T>> {
+    options = options || {};
+    let ids = new Set<string>();
+    return new Observable<StreamChunk<T>>(observer => {
+      options = options || {};
+
+      const streams = new Set<ChangeStream>();
+      const pipeline: object[] = [];
+      const sort = new Subject<any>();
+      let sortSubscription: Subscription;
+
+      if (options.sort) {
+        const sortedKeys = Object.keys(options.sort);
+        sortSubscription = sort
+          .pipe(
+            filter(change => {
+              if (change.operationType == "update") {
+                // TODO: Check this for subdocument updates
+                const changedKeys: string[] = (change.updateDescription.removedFields || []).concat(
+                  Object.keys(change.updateDescription.updatedFields || {})
+                );
+                return changedKeys.some(k => sortedKeys.indexOf(k) > -1);
+              }
+              return true;
+            }),
+            bufferTime(70),
+            filter(changes => changes.length > 0)
+          )
+          .subscribe(async events => {
+            let syncedIds: string[];
+
+            if (
+              sortedKeys.length == 1 &&
+              sortedKeys[0] == "_id" &&
+              events.length > 1 &&
+              options.sort._id == -1
```

Comment:
@thesayyn What does this if block mean?

Reply:
This check should be removed as well as the other commented lines. 

My initial intention was to make the `_id` sort faster by sorting the _id directly in the process.

Reply:
Ok so let's remove them then.